### PR TITLE
Make AuthKey parsing leaner.

### PR DESCRIPTION
**Pull request by markusthoemmes**
Like in the case of ActivationIds, for AuthKeys we do not care if the user really gives us a UUID. For the system internal it doesn't matter.

### CODE REVIEW

**Comment by markusthoemmes** [comment]
PG5 125 🔵

File: common/scala/src/main/scala/whisk/core/entity/AuthKey.scala [hunk 1]
```diff
@@ -31,14 +27,14 @@ import spray.json.deserializationError
...
  */
 protected[core] class AuthKey private (private val k: (UUID, Secret)) extends AnyVal {
-  def uuid = k._1
-  def key = k._2
+  def uuid: UUID = k._1
+  def key: Secret = k._2
   def revoke = new AuthKey(uuid, Secret())
-  def compact = s"$uuid:$key"
-  override def toString = uuid.toString
+  def compact: String = s"$uuid:$key"
```
Comment (rabbah) [inline common/scala/src/main/scala/whisk/core/entity/AuthKey.scala]: fwiw i tend to skip the signature where it's obvious (ie one liners, here .toString, or s"") but OK for uniformity.

File: common/scala/src/main/scala/whisk/core/entity/AuthKey.scala [hunk 2]
```diff
@@ -64,37 +60,22 @@ protected[core] object AuthKey {
...
+   * @return AuthKey if argument is properly formatted
    * @throws IllegalArgumentException if argument is not well formed
    */
   @throws[IllegalArgumentException]
   protected[core] def apply(str: String): AuthKey = {
-    val (k, v) = split(str)
-    new AuthKey(UUID(k), Secret(v))
-  }
+    val (uuid, secret) = str.split(':').toList match {
+      case k :: v :: _ => (k, v)
```
Comment (rabbah) [inline common/scala/src/main/scala/whisk/core/entity/AuthKey.scala]: trim `k` and `v` (preserve existing split) or treat pedantically (new split)?

File: common/scala/src/main/scala/whisk/core/entity/Secret.scala [hunk 3]
```diff
@@ -33,12 +30,12 @@ import spray.json.deserializationError
...
 protected[core] class Secret private (val key: String) extends AnyVal {
-  protected[core] def asString = toString
-  protected[entity] def toJson = JsString(toString)
-  override def toString = key
+  protected[core] def asString: String = toString
+  protected[entity] def toJson: JsString = JsString(toString)
+  override def toString: String = key
 }
 
-protected[core] object Secret extends ArgNormalizer[Secret] {
```
Comment (rabbah) [inline common/scala/src/main/scala/whisk/core/entity/Secret.scala]: should we open an issue to remove arg normalizer?

File: common/scala/src/main/scala/whisk/core/entity/AuthKey.scala (hunk 1 above)
Comment (markusthoemmes) [reply common/scala/src/main/scala/whisk/core/entity/AuthKey.scala]: IntelliJ complains about no signature on public members, so I lazily fix files throughout.

File: common/scala/src/main/scala/whisk/core/entity/Secret.scala (hunk 3 above)
Comment (markusthoemmes) [reply common/scala/src/main/scala/whisk/core/entity/Secret.scala]: I think it's okay to do it lazily/when it gets in the way.

**Pull request merged**